### PR TITLE
Add maven profile to verify readme

### PR DESCRIPTION
**Pull request by srnagar**
The intent of this PR is to add maven tools to update README files whenever sample code in Java file is updated. This also verifies that README is up-to-date and fails the build if the README file does not match the code sample in Java file.

### CODE REVIEW

**Comment by JimSuplizio** [comment]
@srnagar 
I'm curious as to what the process is here. Does the README get updated with the code sample during a local build, if the developer changes the sample, and then the developer submits any changed README with their changes?

**Comment by srnagar** [comment]
@JimSuplizio Yes, here's the process:
- the developers will locally edit ReadmeSamples.java to update/add code snippets
- run `mvn package -Preadme-codesnippets -f pom.client.xml` to automatically update the README.md files with the new code samples
- check-in the java and readme files for PR
- pipeline then runs `verify` on this PR and
- if the readme doesn't match the java files, an error is thrown and the build fails. 

This is to catch cases where developers updated the `ReadmeSamples.java` file only and did not update the readme files.

File: pom.client.xml [hunk 1]
```diff
@@ -972,6 +972,150 @@
...
+                <phase>prepare-package</phase>
+                <goals>
+                  <goal>exec</goal>
+                </goals>
+                <configuration>
+                  <executable>npx</executable>
+                  <arguments>
+                    <argument>embedme</argument>
+                    <argument>sdk/*/azure-*/README.md</argument>
+                  </arguments>
```
Comment (hemanttanwar) [inline pom.client.xml]: Would  this also cover the README in modules. For example 

sdk/eventhubs/README.md
sdk/eventhubs/azure-messaging-eventhubs/README.md

Comment (srnagar) [reply pom.client.xml]: As discussed, the code samples should only be at the module level and shouldn't be at sdk/*/ level. So, not including those READMEs here.

**Pull request merged**